libpam/pam_handlers.c

```diff
@@ -572,93 +584,166 @@ int _pam_init_handlers(pam_handle_t *pamh)
     return PAM_SUCCESS;
 }
 
+static int _pam_buffer_add(struct line_buffer *buffer, char *start, char *end)
+{
+    size_t len = end - start;
+
+    D(("assembled: [%zu/%zu] '%s', adding [%zu] '%s'",
+	buffer->len, buffer->size,
+	buffer->assembled == NULL ? "" : buffer->assembled, len, start));
```

Comment:
This assumes that `strlen(start) == len`.

Reply:
The `len` output in `[%zu]` does not assume that `strlen(start) == len`. Instead, if `len` is larger than `strlen(start)`, then a NUL byte has been encountered in the input line. We want to add it to our parsed line as well so the configuration line interpreter will end and this NUL byte without stopping the parser to search for escaped newlines afterwards.